### ISSUE TITLE
Unable to dump/load custom classes since langchain_core 0.0.13

### ISSUE THREAD

**Issue by simonschmidt** [langchain-ai/langchain#14589]
### System Info

python: 3.11
langchain: 0.0.347-0.0.349
langchain_core: 0.0.12, 0.0.13

### Who can help?

@hwchase17 

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

```python
import langchain_core.load

class Custom(langchain_core.load.Serializable):
    @classmethod
    def is_lc_serializable(cls) -> bool:
        return True

out = langchain_core.load.dumps(Custom())

langchain_core.load.loads(
    out,
    valid_namespaces=["langchain", "__main__"],
)
```

### Expected behavior

I'm expecting it's possible to make a custom class serializable, but since langchain_core 0.0.13 the `valid_namespaces` is effectively ignored as it relies on a whitelist of what can be serialized `SERIALIZABLE_MAPPING`.

So I get the error:

> ValueError: Trying to deserialize something that cannot be deserialized in current version of langchain-core: ('__main__', 'Custom')

Triggered in [load.py#L68](https://github.com/langchain-ai/langchain/blob/v0.0.349/libs/core/langchain_core/load/load.py#L68)

--- 

I'm not sure if serialization was ever intended to be part of the public API, but I've found it convenient to be able to make my custom parts of chains also abide by the serialization protocol and still be able to dump/load my chains

**Comment by hwchase17** [comment]
thanks for flagging @simonschmidt 

this was a change we recently made in order to be a bit more thoughtful about serialization. it wasnt intended (yet) to be part, but as we release a 0.1 release (today) we will highlight it more

I opened this pr as a potential fix: https://github.com/langchain-ai/langchain/pull/14606

basically, if the path is in one of the known namespaces (the langchain ones) then we'll be constrained to the specific mapping we hardcode. otherwise, we'll use the old way

I confirmed that this fixes your error. TBH, we may change some of the logic to be even more permissive in the future, but I think this is safest/fine for now

thoughts?
